From 3e2601903909c2f5ab2ad3bdd6d2edc30454819b Mon Sep 17 00:00:00 2001
From: Robert Dobrowolski <robert.dobrowolski@intel.com>
Date: Tue, 23 Feb 2016 14:35:18 +0100
Subject: [PATCH 39/41] usb: gadget: Fix for adb issue during S3 stress test

During S3 stress tests, adb is lost permanently after multiple
S0->S3->S0 transitions. dwc3_gadget_disconnect_interrupt contains
condition that dwc3_disconnect_gadget(dwc) is being run only when USB
speed is known. disconnect_interrupt should always disconnect the
gadget.

This is partial revert to commit:
07f8f284953ecb8520fb0625ab522b8890aef4b3

Change-Id: Ibdc7195e80d26a50b1f15cb8bb1e8491da8d2e32
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31683
Signed-off-by: Robert Dobrowolski <robert.dobrowolski@intel.com>
Reviewed-on: https://android.intel.com:443/473534
Reviewed-on: https://android.intel.com:443/477325
---
 drivers/usb/dwc3/gadget.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 1357bdc..e7bed9b 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -2331,8 +2331,7 @@ static void dwc3_gadget_disconnect_interrupt(struct dwc3 *dwc)
 	reg &= ~DWC3_DCTL_INITU2ENA;
 	dwc3_writel(dwc->regs, DWC3_DCTL, reg);
 
-	if (dwc->gadget.speed != USB_SPEED_UNKNOWN)
-		dwc3_disconnect_gadget(dwc);
+	dwc3_disconnect_gadget(dwc);
 	dwc->start_config_issued = false;
 
 	dwc->gadget.speed = USB_SPEED_UNKNOWN;
-- 
1.9.1

